thirdeye/thirdeye-frontend/app/pods/self-serve/create-alert/controller.js

```diff
@@ -4,29 +4,647 @@
+   * @param {Number} metricId - Id of selected metric to graph
+   * @return {undefined}
+   */
+  triggerGraphFromMetric(metricId) {
+    const maxTime = this.get('maxTime');
+    const currentEnd = moment(maxTime).isValid()
+      ? moment(maxTime).valueOf()
+      : moment().subtract(1,'day').endOf('day').valueOf();
+    const granularity = this.get('selectedGranularity') || this.get('granularities.firstObject');
+    const currentStart = moment(currentEnd).subtract(1, 'months').valueOf();
+    const baselineStart = moment(currentStart).subtract(1, 'week').valueOf();
+    const baselineEnd = moment(currentEnd).subtract(1, 'week');
+    const { id } = metricId;
+
+    const graphConfig = {
+      id,
+      dimension: 'All',
+      currentStart,
+      currentEnd,
+      baselineStart,
+      baselineEnd,
+      granularity,
+    };
+
+    this.set('graphConfig', graphConfig);
+    this.set('selectedGranularity', granularity);
+
+    this.fetchAnomalyGraphData(this.get('graphConfig')).then(metricData => {
+      this.set('isMetricSelected', true);
+      this.set('selectedMetric', metricData);
+      this.set('loading', false);
+    });
+  },
+
+  /**
+   * Sends a request to begin advanced replay for a metric. The replay will fetch new
+   * time-series data based on user-selected sensitivity settings.
+   * @method triggerReplay
+   * @param {Object} functionObj - the alert object
+   * @param {Object} groupObj - the alert notification group object
+   * @param {Number} newFuncId - the id for the newly created function (alert)
+   * @return {Ember.RSVP.Promise}
+   */
+  triggerReplay(functionObj, groupObj, newFuncId) {
+    const startTime = moment().subtract(1,'month').endOf('day').format("YYYY-MM-DD");
+    const startStamp = moment().subtract(1,'day').endOf('day').valueOf();
+    const endTime = moment().subtract(1,'day').endOf('day').format("YYYY-MM-DD");
+    const endStamp = moment().subtract(1,'month').endOf('day').valueOf();
+    const granularity = this.get('graphConfig.granularity').toLowerCase();
+    const postProps = {
+      method: 'POST',
+      headers: { 'Content-Type': 'Application/Json' }
+    };
+    const replayApi = {
+      base: '/api/detection-job',
+      minute: `/replay/singlefunction?functionId=${newFuncId}&start=${startTime}&end=${endTime}`,
+      hour: `/${newFuncId}/replay?start=${startTime}&end=${endTime}`,
+      day: `/replay/function/${newFuncId}?start=${startTime}&end=${endTime}&goal=1.0&evalMethod=F1_SCORE&includeOriginal=false&tune=\{"pValueThreshold":\[0.001,0.005,0.01,0.05\]\}`,
+      reports: `/thirdeye/email/generate/metrics/${startStamp}/${endStamp}?metrics=${functionObj.metric}&subject=Your%20Metric%20Has%20Onboarded%20To%20Thirdeye&from=thirdeye-noreply@linkedin.com&to=${groupObj.recipients}&smtpHost=email.corp.linkedin.com&smtpPort=25&includeSentAnomaliesOnly=true&isApplyFilter=true`
+    };
+    let gkey = '';
+
+    if (granularity.includes('minute')) { gkey = 'minute'; }
```

Comment:
would something like this work? 
```
const allowedGranularity = ['minute', 'hour', 'day'];
gkey = allowedGranularity.includes(granularity) ? granularity : '';
```

Or we could create a `const GRANULARITY_MAPPING = {5_minutes: 'minutes', ...}`